File: file-service/src/main/java/com/comup/file/service/entity/pk/FileInfoPK.java
package com.comup.file.service.entity.pk;

import javax.persistence.Embeddable;
import java.io.Serializable;

@Embeddable
public class FileInfoPK implements Serializable {
    private static final long serialVersionUID = -8246906426037035118L;

    private String scheme;
    private String host;
    private Integer port;
    private String path;
    private String fileName;

    public String getScheme() {
        return scheme;
    }

    public void setScheme(String scheme) {
        this.scheme = scheme;
    }

    public String getHost() {
        return host;
    }

    public void setHost(String host) {
        this.host = host;
    }

    public Integer getPort() {
        return port;
    }

    public void setPort(Integer port) {
        this.port = port;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }
}

File: file-service/src/main/kotlin/com/comup/file/service/entity/pk/FileInfoPK.kt
package com.comup.file.service.entity.pk

import com.fasterxml.jackson.annotation.JsonInclude
import java.io.Serializable
import javax.persistence.Embeddable

@Embeddable
@com.fasterxml.jackson.annotation.JsonInclude(JsonInclude.Include.NON_EMPTY)
data class FileInfoPK(
        var scheme: String? = "",
        var host: String? = "",
        var port: Int? = 0,
        var path: String? = "",
        var fileName: String? = ""
) : Serializable {
}